pkg/ipamd/ipamd.go

```diff
@@ -1102,18 +1100,66 @@ func getMinimumIPTarget() int {
 	return noMinimumIPTarget
 }
 
-func filterUnmanagedENIs(enis []awsutils.ENIMetadata) ([]awsutils.ENIMetadata, int) {
-	numFiltered := 0
+// filterUnmanagedENIs filters out ENIs marked with the "node.k8s.amazonaws.com/no_manage" tag
+func (c *IPAMContext) filterUnmanagedENIs(enis []awsutils.ENIMetadata) []awsutils.ENIMetadata {
+	noManageENIs := make([]string, 0, len(enis))
+	// Update filter if needed
+	if c.needTagRefresh() {
```

Comment:
Isn't there is a race condn if we invoke nodeIpPoolReconcile and if the ENI is just attached then CNI will try to add/remove IPs from the one which it shouldnt manage right?

Reply:
Yes. I've rewritten this to not allow any tag refreshing. The ENIs will have to be tagged before they are attached. Another option would be to actively filter for the ones we tag ourselves.. 🤔